re_data_ui/src/partials/Graph.tsx

```diff
@@ -0,0 +1,328 @@
+};
+
+const generateGraph = (overview: OverviewData, modelName?: string | null) => {
+  const graph: IGraph = {
+    nodes: [],
+    edges: [],
+  };
+  if (!overview.graph) {
+    return graph;
+  }
+  const { dbtMapping } = overview;
+  const dbtNodes = overview.graph.nodes;
+  const dbtSources = overview.graph.sources;
+
+  const allNodes = { ...dbtNodes, ...dbtSources };
+
+  if (modelName) {
+    const { parent_map: parentNodes, child_map: childNodes } = overview.graph;
+    const modelTitle = dbtMapping[modelName];
+
+    // get all the parents and the child nodes of the model name;
+    const modelParentNodes = parentNodes[modelTitle] || [];
+    const modelChildNodes = childNodes[modelTitle] || [];
+
+    // draw the model node
+    // draw the parent nodes and connect it to the node using an edge
+    // draw the child nodes and connect it to the node using an edge
+
+    const details = allNodes[modelTitle];
+    const modelId = `${details.database}.${details.schema}.${details.name}`.toLowerCase();
+    graph.nodes.push({
+      id: modelId,
+      label: details.name,
+      shape: 'box',
+      color: resourceTypeColors[details.resource_type],
+    });
+
+    if (modelParentNodes.length) {
+      modelParentNodes.forEach((parent) => {
+        const parentDetails = allNodes[parent];
+        const {
+          database, schema, name,
+          resource_type: resourceType,
+        } = parentDetails;
+        if (supportedResTypes.has(resourceType)) {
+          const parentModelId = `${database}.${schema}.${name}`.toLowerCase();
+          graph.nodes.push({
+            id: parentModelId,
+            label: name,
+            shape: 'box',
+            color: resourceTypeColors[resourceType],
+          });
+
+          const edge: VisEdge = {
+            from: parentModelId,
+            to: modelId,
+            arrows: 'to',
+          };
+          graph.edges.push(edge);
+        }
+      });
+    }
+    if (modelChildNodes.length) {
+      modelChildNodes.forEach((child) => {
+        const childDetails = allNodes[child];
+        const {
+          database, schema, name,
+          resource_type: resourceType,
+        } = childDetails;
+        if (supportedResTypes.has(resourceType)) {
+          const childModelId = `${database}.${schema}.${name}`.toLowerCase();
+          graph.nodes.push({
+            id: childModelId,
+            label: name,
+            shape: 'box',
+            color: resourceTypeColors[resourceType],
+          });
+
+          const edge: VisEdge = {
+            from: modelId,
+            to: childModelId,
+            arrows: 'to',
+          };
+          graph.edges.push(edge);
+        }
+      });
+    }
+  } else {
+    Object.entries(allNodes).forEach(([, details]) => {
+      if (supportedResTypes.has(details.resource_type) && details.package_name !== 're_data') {
+        const modelId = `${details.database}.${details.schema}.${details.name}`.toLowerCase();
```

Comment:
we can refactor this logic into a function since we're using it in more than one place. what do you think?

Reply:
i.e generating the identifier

Reply:
okay, i'll convert it into an helper function